packages/xrpl/src/client/index.ts

```diff
@@ -489,6 +496,7 @@ class Client extends EventEmitter<EventTypes> {
       })
       this.networkID = response.result.info.network_id ?? undefined
       this.buildVersion = response.result.info.build_version
+      this.apiVersion = parseInt(this.buildVersion.split('.')[0], 10)
```

Comment:
I am not a fan of this.  This will be a major breaking change.  Anyone using public infrastructure will be one api 2.0.  also without complete 2.0 type support this should not be done.

Reply:
@ckniffen That's fair. What do you recommend doing? Should we just leave it as it (defaults to rippled v1 API) and the user supplies what API version to use?

Reply:
Yes default to not sending one and if they set version to 2 then we derive different types